From 4ef5ada0be476033d8d74140fa896a36ff9c4326 Mon Sep 17 00:00:00 2001
From: Zhang Jiejing <jiejing.zhang@freescale.com>
Date: Fri, 15 Apr 2011 10:02:33 +0800
Subject: [PATCH 0110/2463] ENGR00137340-2 MX53_SMD: add headphone detection
 support.

add headphone gpio in imx sound card platform data.
delete old method of headphone detection codec.

Signed-off-by: Zhang Jiejing <jiejing.zhang@freescale.com>
---
 arch/arm/mach-mx5/board-mx53_smd.c |    6 ++----
 1 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-mx5/board-mx53_smd.c b/arch/arm/mach-mx5/board-mx53_smd.c
index 281320b..4780d48 100755
--- a/arch/arm/mach-mx5/board-mx53_smd.c
+++ b/arch/arm/mach-mx5/board-mx53_smd.c
@@ -580,10 +580,6 @@ static int smd_sgtl5000_init(void)
 {
 	smd_audio_data.sysclk = 22579200;
 
-	/* headphone_det_b */
-	gpio_request(MX53_SMD_HEADPHONE_DEC, "headphone-dec");
-	gpio_direction_input(MX53_SMD_HEADPHONE_DEC);
-
 	/* Enable OSC_CKIH1_EN for audio */
 	gpio_request(MX53_SMD_OSC_CKIH1_EN, "osc-en");
 	gpio_direction_output(MX53_SMD_OSC_CKIH1_EN, 1);
@@ -595,6 +591,8 @@ static struct mxc_audio_platform_data smd_audio_data = {
 	.src_port = 2,
 	.ext_port = 5,
 	.init = smd_sgtl5000_init,
+	.hp_gpio = MX53_SMD_HEADPHONE_DEC,
+	.hp_active_low = 1,
 };
 
 static struct platform_device smd_audio_device = {
-- 
1.7.7.4

